features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/APIClientFactory.js

```diff
@@ -43,22 +45,30 @@ class APIClientFactory {
      * @param {Object} environment
      * @returns {APIClient} APIClient object for the environment
      */
-    getAPIClient(environment) {
+    getAPIClient(environment, clientType) {
         const {
             label,
         } = environment;
         if (label === undefined) {
             throw new Error('Environment label is undefined, Please provide'
                 + 'a valid environment object with keys (host,label & loginTokenPath)');
         }
-        let apiClient = this._APIClientMap.get(environment.label);
-
+        let apiClient;
+        if (clientType === Utils.CONST.API_CLIENT) {
```

Comment:
Shall we combine this check and the below check (if conditions) 

```
if (clientType === Utils.CONST.API_CLIENT) {
```
and update the `_APIClientMap` if there is no client afterward

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/9372/commits/929acf282539bdf1dd875f64e98ceec5152609c3